### PR TITLE
Use values from personalizationOption, hardcoded values reset selecti…

### PR DESCRIPTION
**Pull request by ivan-bui**
…on on cell reuse

### CODE REVIEW

**Comment by ivan-bui** [comment]
why the builds failing? :(

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m [hunk 1]
```diff
@@ -59,9 +61,26 @@ - (void)registerNibs {
...
+
+- (void)initDefaultFontAndColorOptions {
+    self.fontNames = [self defaultFontNames];
+    self.colors = [self defaultColorOptions];
+}
+
+- (NSArray<UIColor *> *)defaultColorOptions {
+    UIColor *navyColor = [UIColor colorWithHexString:@"#2D3790" withAlpha:1.f];
+    UIColor *blackColor = [UIColor colorWithHexString:@"#261B15" withAlpha:1.f];
+    UIColor *redColor = [UIColor colorWithHexString:@"#FF4B47" withAlpha:1.f];
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m]: Don't use the hex, just do 
```
UIColor *blackColor = [UIColor flatBlackColor];
UIColor *blueColor = [UIColor flatBlueColor];
etc
```

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m [hunk 2]
```diff
@@ -22,8 +22,10 @@ @interface NoteCustomizationCollectionViewCell () <CheckboxViewDelegate>
 @property (weak, nonatomic) IBOutlet UICollectionView *colorsCollectionView;
 @property (weak, nonatomic) IBOutlet UICollectionView *fontsCollectionView;
 
-@property (nonatomic) NSMutableSet<NSString *> *fontNames;
-@property (nonatomic) NSMutableSet<UIColor *> *colors;
+@property (nonatomic) NSArray<NSString *> *fontNames;
+@property (nonatomic) NSArray<UIColor *> *colors;
+@property (nonatomic) NSMutableArray<NSString *> *mutableFontNames;
+@property (nonatomic) NSMutableArray<UIColor *> *mutableColors;
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m]: Change name to selectedFontNames, selectedColors

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m [hunk 3]
```diff
@@ -22,8 +22,10 @@ @interface NoteCustomizationCollectionViewCell () <CheckboxViewDelegate>
 @property (weak, nonatomic) IBOutlet UICollectionView *colorsCollectionView;
 @property (weak, nonatomic) IBOutlet UICollectionView *fontsCollectionView;
 
-@property (nonatomic) NSMutableSet<NSString *> *fontNames;
-@property (nonatomic) NSMutableSet<UIColor *> *colors;
+@property (nonatomic) NSArray<NSString *> *fontNames;
+@property (nonatomic) NSArray<UIColor *> *colors;
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m]: Change name to colorOptions

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m [hunk 4]
```diff
@@ -22,8 +22,10 @@ @interface NoteCustomizationCollectionViewCell () <CheckboxViewDelegate>
 @property (weak, nonatomic) IBOutlet UICollectionView *colorsCollectionView;
 @property (weak, nonatomic) IBOutlet UICollectionView *fontsCollectionView;
 
-@property (nonatomic) NSMutableSet<NSString *> *fontNames;
-@property (nonatomic) NSMutableSet<UIColor *> *colors;
+@property (nonatomic) NSArray<NSString *> *fontNames;
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m]: Change to fontOptions

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m (hunk 3 above)
Comment (rbui) [reply VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/NoteCustomizationCollectionViewCell.m]: You actually don't even need properties for these, just use the methods directly

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/PhotoCustomizationCollectionViewCell.m [hunk 5]
```diff
@@ -78,21 +78,30 @@ - (void)setupWithPersonalizationConfig:(PersonalizationConfigModel *)config {
...
+    [self.maxSearchResultSliderView setCurrentValue:[[self valueOfPersonalizationOption:@"maxSearchResults"] intValue]];
     
     self.suggestedSearchTerms = @[@"Meme", @"Funny", @"Thanks", @"Pattern", @"I Owe You"];
     self.mutableSuggestedSearchTerms = [self.suggestedSearchTerms mutableCopy];
-    [((PhotoPersonalizationOption *)self.config.personalizationOption) setValue:self.mutableSuggestedSearchTerms forKey:@"suggestedSearchTerms"];
 }
 
+- (id)valueOfPersonalizationOption:(NSString *)property {
+    return [self.config.personalizationOption valueForKey:property];
+}
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/PhotoCustomizationCollectionViewCell.m]: can delete

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/RecipientCustomizationCollectionViewCell.m [hunk 6]
```diff
@@ -55,22 +55,26 @@ - (void)setupWithPersonalizationConfig:(PersonalizationConfigModel *)config {
...
                            toggleOptions:@[[ToggleOption optionWithText:@"Users from app" font:nil textColor:nil backgroundColor:nil],
                                            [ToggleOption optionWithText:@"Users from server" font:nil textColor:nil backgroundColor:nil]]
-                           selectedIndex:1
+                           selectedIndex:[self valueOfPersonalizationOption:@"userGroupsVouchersCanBeSentTo"] ? 0 : 1
                                 delegate:self];
 }
 
+- (id)valueOfPersonalizationOption:(NSString *)property {
+    return [self.config.personalizationOption valueForKey:property];
+}
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/RecipientCustomizationCollectionViewCell.m]: can delete. also, if you are implementing this method in every file, you should consider making a category or adding the method to the class

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/WrappingCustomizationCollectionViewCell.m [hunk 7]
```diff
@@ -58,10 +58,12 @@ - (void)setupConfigurationControls {
...
-    [self.config.personalizationOption setValue:@(YES) forKey:@"canSearch"];
-    [self.allowUploadPhotosCheckboxView setupWithDescription:@"Allow User To Upload Image" isChecked:YES delegate:self];
-    [self.config.personalizationOption setValue:@(YES) forKey:@"canUploadPhotos"];
+    [self.allowSearchCheckboxView setupWithDescription:@"Allow Google Image Search" isChecked:[[self valueOfPersonalizationOption:@"canSearch"] boolValue] delegate:self];
+    [self.allowUploadPhotosCheckboxView setupWithDescription:@"Allow User To Upload Image" isChecked:[[self valueOfPersonalizationOption:@"canUploadPhotos"] boolValue] delegate:self];
+}
+
+- (id)valueOfPersonalizationOption:(NSString *)property {
+    return [self.config.personalizationOption valueForKey:property];
 }
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/WrappingCustomizationCollectionViewCell.m]: delete

**Pull request merged**